src/translator/definitions.h

```diff
@@ -21,6 +21,24 @@ template <class T, typename... Args> UPtr<T> UNew(Args &&... args) {
 
 template <class T> UPtr<T> UNew(UPtr<T> p) { return UPtr<T>(p); }
 
+class MemoryGift{
+private:
+  const void *memory_;
+  const size_t memorySize_;
+public:
+  MemoryGift(const void *memory,const size_t memorySize)
+  :memory_(memory),memorySize_(memorySize){}
+  const void* data() const{
+    return memory_;
+  }
+  const size_t length() const{
+    return memorySize_;
+  }
+  void early_free(){
+    free((char *)memory_);
+  }
```

Comment:
> And it could be a method to some blob class.

Weren't the loads supposed to be consolidated here per previous review?

Reply:
Not sure what you're asking.  It's Mozilla's problem to make the (aligned) gift.

Reply:
replacement server (if deployed as a service and has a swap models operation) can also use the binary files correct, if load speeds are improved. It's also my problem to make `MemoryGift` in that case?

I'm overall confused about who owns MemoryGift. Does mozilla swap in swap out models in which case they do. If we are, there are a lot of leaks in the current code, is there not - due to ownership issues.

Reply:
This is resolved in another thread. @qianqianzhu wanted to keep IO and the struct separate. Reopen if not the case.